Add(Managed)Dependency - throws exception if current `dependencies` only contains comments

https://github.com/openrewrite/rewrite/blob/76dab52bae9a2431f43af8460c0d68b2c5df4150/rewrite-maven/src/test/java/org/openrewrite/maven/AddDependencyTest.java#L526-L568

This test case currently fails with an exception:
```
Caused by: org.openrewrite.internal.RecipeRunException: java.lang.NullPointerException: Cannot read field "value" because "anotherFloat" is null
	at org.openrewrite.TreeVisitor.visit(TreeVisitor.java:329)
	at org.openrewrite.TreeVisitor.visitAndCast(TreeVisitor.java:361)
	at org.openrewrite.xml.XmlVisitor.lambda$visitTag$2(XmlVisitor.java:91)
	at org.openrewrite.internal.ListUtils.map(ListUtils.java:176)
	at org.openrewrite.xml.XmlVisitor.visitTag(XmlVisitor.java:91)
	at org.openrewrite.xml.AddToTagVisitor.visitTag(AddToTagVisitor.java:88)
	at org.openrewrite.xml.tree.Xml$Tag.acceptXml(Xml.java:430)
	at org.openrewrite.xml.tree.Xml.accept(Xml.java:50)
	at org.openrewrite.TreeVisitor.visit(TreeVisitor.java:278)
	at org.openrewrite.TreeVisitor.visitAndCast(TreeVisitor.java:361)
	at org.openrewrite.xml.XmlVisitor.visitDocument(XmlVisitor.java:71)
	at org.openrewrite.xml.tree.Xml$Document.acceptXml(Xml.java:151)
	at org.openrewrite.xml.tree.Xml.accept(Xml.java:50)
	at org.openrewrite.TreeVisitor.visit(TreeVisitor.java:278)
	at org.openrewrite.TreeVisitor.visit(TreeVisitor.java:314)
	at org.openrewrite.TreeVisitor.visit(TreeVisitor.java:314)
	at org.openrewrite.TreeVisitor.visitNonNull(TreeVisitor.java:197)
	at org.openrewrite.maven.AddDependency$2.visitDocument(AddDependency.java:228)
	at org.openrewrite.maven.AddDependency$2.visitDocument(AddDependency.java:193)
	at org.openrewrite.xml.tree.Xml$Document.acceptXml(Xml.java:151)
	at org.openrewrite.xml.tree.Xml.accept(Xml.java:50)
	at org.openrewrite.TreeVisitor.visit(TreeVisitor.java:278)
	at org.openrewrite.TreeVisitor.visit(TreeVisitor.java:184)
	at org.openrewrite.ScanningRecipe$1.visit(ScanningRecipe.java:112)
	at org.openrewrite.ScanningRecipe$1.visit(ScanningRecipe.java:94)
	at org.openrewrite.RecipeScheduler$RecipeRunCycle.lambda$editSources$3(RecipeScheduler.java:231)
	at io.micrometer.core.instrument.AbstractTimer.recordCallable(AbstractTimer.java:147)
	at org.openrewrite.table.RecipeRunStats.recordEdit(RecipeRunStats.java:56)
	at org.openrewrite.RecipeScheduler$RecipeRunCycle.lambda$editSources$4(RecipeScheduler.java:228)
	... 14 more
Caused by: java.lang.NullPointerException: Cannot read field "value" because "anotherFloat" is null
	at java.base/java.lang.Float.compareTo(Float.java:923)
	at org.openrewrite.maven.internal.InsertDependencyComparator.compare(InsertDependencyComparator.java:67)
	at org.openrewrite.maven.internal.InsertDependencyComparator.compare(InsertDependencyComparator.java:25)
	at org.openrewrite.xml.AddToTagVisitor.visitTag(AddToTagVisitor.java:72)
	at org.openrewrite.xml.tree.Xml$Tag.acceptXml(Xml.java:430)
	at org.openrewrite.xml.tree.Xml.accept(Xml.java:50)
	at org.openrewrite.TreeVisitor.visit(TreeVisitor.java:278)
	... 42 more
```
## Are you interested in [contributing a fix to OpenRewrite](https://github.com/openrewrite/.github/blob/main/CONTRIBUTING.md#contributing-fixes)?
Maybe; not a super-critical issue for me

--- a/rewrite-maven/build.gradle.kts
+++ b/rewrite-maven/build.gradle.kts
@@ -38,7 +38,6 @@ dependencies {
     implementation("org.apache.commons:commons-text:latest.release")
 
     testImplementation(project(":rewrite-test"))
-    testImplementation("org.junit-pioneer:junit-pioneer:2.0.0")
     testImplementation("com.squareup.okhttp3:mockwebserver:4.+")
     testImplementation("com.squareup.okio:okio-jvm:3.0.0")
     testImplementation("org.mapdb:mapdb:latest.release")

--- a/rewrite-maven/src/main/java/org/openrewrite/maven/internal/InsertDependencyComparator.java
+++ b/rewrite-maven/src/main/java/org/openrewrite/maven/internal/InsertDependencyComparator.java
@@ -64,7 +64,8 @@ public class InsertDependencyComparator implements Comparator<Content> {
 
     @Override
     public int compare(Content o1, Content o2) {
-        return positions.get(o1).compareTo(positions.get(o2));
+        Float anotherFloat = positions.get(o2);
+        return anotherFloat == null ? 1 : positions.get(o1).compareTo(anotherFloat);
     }
 
     private static final Comparator<Xml.Tag> dependencyComparator = (d1, d2) -> {

--- a/rewrite-maven/src/main/java/org/openrewrite/maven/internal/InsertDependencyComparator.java
+++ b/rewrite-maven/src/main/java/org/openrewrite/maven/internal/InsertDependencyComparator.java
@@ -65,7 +65,7 @@ public class InsertDependencyComparator implements Comparator<Content> {
     @Override
     public int compare(Content o1, Content o2) {
         Float anotherFloat = positions.get(o2);
-        return anotherFloat == null ? 1 : positions.get(o1).compareTo(anotherFloat);
+        return anotherFloat == null ? 0 : positions.get(o1).compareTo(anotherFloat);
     }
 
     private static final Comparator<Xml.Tag> dependencyComparator = (d1, d2) -> {

--- a/rewrite-maven/src/test/java/org/openrewrite/maven/AddDependencyTest.java
+++ b/rewrite-maven/src/test/java/org/openrewrite/maven/AddDependencyTest.java
@@ -19,7 +19,6 @@ import org.intellij.lang.annotations.Language;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.params.ParameterizedTest;
 import org.junit.jupiter.params.provider.ValueSource;
-import org.junitpioneer.jupiter.ExpectedToFail;
 import org.openrewrite.Issue;
 import org.openrewrite.internal.lang.Nullable;
 import org.openrewrite.java.JavaParser;
@@ -551,13 +550,13 @@ class AddDependencyTest implements RewriteTest {
                         <artifactId>my-app</artifactId>
                         <version>1</version>
                         <dependencies>
+                            <!-- my cool dependencies section -->
+                            <!-- etc -->
                             <dependency>
                                 <groupId>com.google.guava</groupId>
                                 <artifactId>guava</artifactId>
                                 <version>29.0-jre</version>
                             </dependency>
-                            <!-- my cool dependencies section -->
-                            <!-- etc -->
                         </dependencies>
                     </project>
                 """

--- a/rewrite-maven/src/test/java/org/openrewrite/maven/AddDependencyTest.java
+++ b/rewrite-maven/src/test/java/org/openrewrite/maven/AddDependencyTest.java
@@ -524,7 +524,6 @@ class AddDependencyTest implements RewriteTest {
     }
 
     @Test
-    @ExpectedToFail
     @Issue("https://github.com/openrewrite/rewrite/issues/3458")
     void addDependencyOopsAllComments() {
         rewriteRun(
@@ -552,13 +551,13 @@ class AddDependencyTest implements RewriteTest {
                         <artifactId>my-app</artifactId>
                         <version>1</version>
                         <dependencies>
-                            <!-- my cool dependencies section -->
-                            <!-- etc -->
                             <dependency>
                                 <groupId>com.google.guava</groupId>
                                 <artifactId>guava</artifactId>
                                 <version>29.0-jre</version>
                             </dependency>
+                            <!-- my cool dependencies section -->
+                            <!-- etc -->
                         </dependencies>
                     </project>
                 """

